### ISSUE TITLE
Using the "%" (percent) wildcard with the like operator in name queries

### ISSUE THREAD

**Issue by fipwmaqzufheoxq92ebc** [eikek/docspell#1114]
Hi, 
I've just started using docspell, and noticed that the "%" can be used as wildcard instead of the "*" asterisk.
I think that's because the percent sign is used as wildcard in SQL LIKE queries (e.g. [mariadb](https://mariadb.com/kb/en/like/), [postgres](https://www.postgresql.org/docs/current/functions-matching.html#FUNCTIONS-LIKE) ) and isn't escaped by docspell.
The "%" is more powerful than the asterisk, as it also can be used in between ("t%t" matches "test").

The "_" (underscore), which can be used to match any single character, also works.

(Tested only on PostgreSQL for the name and source fields).

I think both are more a feature than a bug and should be documented.
And the asterisk, as it can be replaced by "%", should be removed (or deprecated)

**Comment by eikek** [comment]
> > it ties the search function to SQL
> 
> Do you mean because of the usage of the same characters? Then what about using "_" (or some other character), and replace it for SQL (as it is done now for "_")

Yes partly and because it is passed through to SQL unnoticed without being explicitely "authorized". 

> > it allows for maybe slow queries too easily
> 
> I'm unsure if there would be a real performance impact.
> 
> ```
> name:"%h%" name:"%e%" name:"%l%" name:"%l%" name:"%o%"
> ```
> 
> (Results: 311) vs
> 
> ```
> name:"%h%e%l%l%o%"
> ```
> 
> (Results: 89) vs
> 
> ```
> name:"%hello%"
> ```
> 
> (Results: 0) I have 669 items in docspell. The first two queries take around 30-50ms for /searchStats, and 110-140ms for /search. Queries filtered by Tags, or no filter, have similar timings.
> 
> The last query takes around 20-40ms for /searchStats, and 10-15ms for /search.
> 
> If there are more items, it may be slower, but how many items/how much slower? And how many people have such many items and do such queries?

Are these db timings or request timings? You're right, it is totally possible to create  slow queries without using `%`. But it makes it even easier, not?

> > Do you need this "feature"?
> 
> I think all of the queries I used it for could be replaced a separate condition for each word (e.g. `name:"%hello%" name:"%test%"`)
> 
> But it may be useful for searching sentences.
> 
> Personally, I would prefer keeping it as feature as long as there aren't any problems, but wouldn't mind if it's removed.

I had hoped that using fulltext search is easier when searching like this. But I'm also fine with allowing wildcards in between words. I then would use the `*` to be allowed everywhere and replace it with `%` for sql. The `%` sign is not so common (if you know sql, it's fine and makes sense of course) - to me the `*` is more common.

**Comment by eikek** [comment]
Ok, thanks for the answer! I'll look into fulltext performance, normally if it is fulltext only, then it shouldn't be very much slower
